### ISSUE TITLE
.groupBy method does not recognise model column alias

### ISSUE THREAD

**Issue by dmvianna** [balderdashy/waterline#1173]
I have noticed that both PostgreSQL column names and and Waterfront column name aliases, such as

```js
        region: {
            type: "string",
            columnName: 'operations_region_code',
        },
```

are accepted indiscriminately by `.find`. However, when I chain 

```js
        Model.find({select:['region']}).groupBy(['region'])
```
I get undefined. On the other hand, if I chain

```js
        Model.find({select:['region']}).groupBy(['operations_region_code'])
```
I get the expected response.

**Comment by sailsbot** [comment]
Thanks for posting, @dmvianna.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Marked as bug– reopening.

**Comment by TylerBre** [comment]
+1

I encountered this but when trying to get some nested attributes to work

It seems that if there is an association attribute waterline doesn't play well with data manipulation
